modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/openapi/contributor/ExtensionsOpenAPIContributor.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.vulcan.internal.openapi.contributor;
+
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.odata.entity.EntityField;
+import com.liferay.portal.odata.entity.EntityModel;
+import com.liferay.portal.vulcan.openapi.OpenAPIContext;
+import com.liferay.portal.vulcan.openapi.contributor.OpenAPIContributor;
+import com.liferay.portal.vulcan.resource.EntityModelResource;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.MultivaluedHashMap;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Magdalena Jedraszak
+ */
+@Component(service = OpenAPIContributor.class)
+public class ExtensionsOpenAPIContributor implements OpenAPIContributor {
+
+	@Override
+	public void contribute(OpenAPI openAPI, OpenAPIContext openAPIContext)
+		throws Exception {
+
+		Map<String, Schema> schemas = openAPI.getComponents(
+		).getSchemas();
+
+		if (schemas.isEmpty()) {
+			return;
+		}
+
+		Map<String, EntityField> entityFieldsMap = getEntityModelFromJaxrs(
+			openAPIContext.getPath());
+
+		if ((entityFieldsMap == null) || entityFieldsMap.isEmpty()) {
+			return;
+		}
+
+		for (Map.Entry<String, Schema> schemaEntry : schemas.entrySet()) {
```

Comment:
You are just using the values, why don't just call the values() method?

Reply:
I think I was but SF shouted to use entries, let me double check though

Reply:
it worked! :D